libs/core/langchain_core/runnables/history.py

```diff
@@ -96,6 +99,7 @@ def __init__(
         input_messages_key: Optional[str] = None,
         output_messages_key: Optional[str] = None,
         history_messages_key: Optional[str] = None,
+        session_history_config_specs: Optional[Sequence[ConfigurableFieldSpec]] = None,
```

Comment:
```suggestion
        history_factory_config: Optional[Sequence[ConfigurableFieldSpec]] = None,
```

Reply:
Don't love either name - both a bit wordy and difficult to figure out what it is at first glance. Maybe `session_history_config`? Maybe history is redundant because it's chat history